### DISCUSSION TITLE
Get status of file.write

### DISCUSSION THREAD

**Discussion by tuskin40** [IfcOpenShell/IfcOpenShell#4056] [General]
I am using IfcOpenShell to merge IFC files. Is there a way I can check what percentage is done or remaining when writing to a file? I want to be able to give the user feedback with regard to how long it will still take to write this big file. To show the user that it is not frozen and that it is still writng.

**Comment by aothms** [comment]
Currently not. From the discussion title it appears you're using Python? You could just do the file.write() in another threading.Thread() and maybe look at os.path.getsize() vs some sort of estimate of expected filesize.

The alternative is that you write one instance at a time.

In recent versions there is ifcopenshell.entity_instance.to_string() which writes a compliant spf serialization to string (`str(ifcopenshell.entity_instance)` is a bit different). That way if you somehow get the header to print (maybe from an empty file) you can get your own file writing with incremental progress.

I realize APIs for this are far from ideal. All help is welcome :)
